test/run_clippy.sh

```diff
@@ -2,6 +2,10 @@
 
 set -eou pipefail
 
+source ./support/ci/shared.sh
```

Comment:
Is there a distinction to be made between what goes into `support/ci` and `.buildkite`?

Reply:
This is something we'll be revisiting - right now things are spread between `support/ci`, `.buildkite`, `test`, etc. I didn't want to tackle that in this PR since it's going to require a bit more juggling.

Reply:
Can we get an issue to track that work? I doubt you'll forget about it, since I'm sure this is pretty near the top of your mind right now, but it'd be good for general board-walking and awareness purposes.

Reply:
Done! https://github.com/habitat-sh/habitat/issues/6227